compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_aie.mlir

```diff
@@ -1,15 +1,23 @@
 // RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(iree-amdaie-lower-to-aie)" --verify-diagnostics %s | FileCheck %s
 
-// CHECK: module
+// expected-error @+1 {{No AMDAIEDevice found in the target attribute configuration}}
 module {
 }
 
 // -----
 
+// CHECK: module
+#executable_target_amdaie_xclbin_fb = #hal.executable.target<"amd-aie", "amdaie-xclbin-fb", {target_device = 8 : i32, ukernels = "none"}>
```

Comment:
I think we should use the text of the target rather tahn `8` ?

Reply:
How to do that?

Reply:
nah I disagree - if you use text you won't be able to interop with eventual C usages of the enum.

Reply:
I think it
1. makes the command line obscure for developers
2. different from every other iree backend that take a string attribute here. 
3. Leaking lower level workings that may change to high level API which will have to change each time.
Anyway looks like it will require changing the device attribute to string and then having a mapping function but you guys can make the call if you like it this way we can leave it.

Reply:
I mean re CLI vs purely IR based I don't have a strong opinion. I was only talking about int vs string for the attribute. 

> different from every other iree backend that take a string attribute here.

Ignorant on that - if conventionally it's a string then I'm slightly more sympathetic

> Leaking lower level workings that may change to high level API which will have to change each time.

I guess that's true but I don't see it changing (you'd have to be a sadist to like reorder an enum :angry:). 

> then having a mapping function

High-level I think what I really don't like is the proliferation/profusion of these banal kinds of things (like enums) that aren't unified/have to be constantly synced up. Like we (via mlir-aie) already have this all over the place because stuff like `WireBundle`, `DMAChannel`, etc that they declared/defined as MLIR attributes when they already existed as perfectly good enums in `aie-rt`. So now we have stuff like [this in a bunch of places](https://github.com/nod-ai/iree-amd-aie/blob/main/runtime/src/iree-amd-aie/aie_runtime/test/DeviceModelTest.cpp#L33-L44). Now you could say "well just promote that to a header" but I'm working in the other direction (removing `WireBundle` and "canonicalizing" on `StrmSwPortType`).

Reply:
> I guess that's true but I don't see it changing (you'd have to be a sadist to like reorder an enum 😠).
Wont it change when you decouple the row/columns ?

Anyway I would feel awkward telling the front end SHARK tem to use `8` if they want phoenix and `9` for strix but its NBD.

Reply:
Based on the discussion here, I updated it to use a `StringAttr` purely for the IR, while keeping the `AMDAIEDevice` as is. Could you guys have another look?

Reply:
Nice, I think this way it is string and is not talking on any extra mapping that Max was concerned about so I love it!

Reply:
> Wont it change when you decouple the row/columns ?

No - the values would stay the same and still be required, just that `partitionStart` and `partitionCols` would _also_ be required.

> Anyway I would feel awkward telling the front end SHARK team to use `8` if they want phoenix and `9` for strix but its NBD.

Fair enough. Anyway Jorn threaded the needle so :partying_face:.